app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,44 @@
+class RestaurantsController < ApplicationController
+  before_action :set_restaurant, only: [:show, :update, :destroy]
+
+  def index
+    @restaurants = Restaurant.all
+
+    render json: @restaurants
+  end
+
+  def show
+    render json: @restaurant
+  end
+
+  def create
+    @restaurant = Restaurant.new(restaurant_params)
```

Comment:
You can also `create!` a restaurant, which does `new` + `save!`, and then rescue an error if it fails. (This is also great, just wanted you to know both ways)